cdci_data_analysis/flask_app/app.py

```diff
@@ -90,8 +84,24 @@ def run_api_parameters():
 
 @app.route("/api/instr-list")
 def run_api_instr_list():
-    instrument_list = InstrumentQueryBackEnd.get_user_specific_instrument_list(app)
-    return instrument_list
+    logger.warning('\nThe endpoint \'/api/instr-list\' is deprecated and you will be automatically redirected to the '
+                   '\'/instr-list\' endpoint. Please use this one in the future.\n')
+
+    bind_host = app.config['conf'].bind_host
+    bind_port = app.config['conf'].bind_port
+
+    parsed_product_url = urlparse(app.config['conf'].products_url)
+
+    if (parsed_product_url.hostname is None and parsed_product_url.port is None) or \
```

Comment:
all is this quite convoluted. Maybe it is possible to use `request.referrer` ?

Reply:
Tried a bit, but it always returns an empty value. Is there something I am missing?

Reply:
maybe `request.url`?

Reply:
`request.url` does not return the bind_host and bind_port, but the url that flask assigns when starting the app.

How could use it in this case?

Reply:
you do not need bind, you need to know URL used to call the service. usually, it can be found. can you investigate how to discover this url?

Reply:
you could use it as a base for redirect. Do make sure it makes sense.
But, what you want, is to return to the user the same URL they used, just replacing the "/api/" part, right? 
It's good to check for clarity that it matches that in the config

Reply:
Yes, basically, replacing the `/api` part

Reply:
It now uses `request.url`, but I am a bit stuck on how to test that `products_url` within the config matches the one used for building `redirection_url`

Reply:
products url is a frontend url. if it's set (like in production and some tests) it should match the redirect with dispatch-data. if not, it should be ignored

Reply:
Checks now if products_url from config is set and is a valid url. Done by using the `validators` library